### PR TITLE
Add `[T; N]` for `Robj`.

### PR DESCRIPTION
**Pull request by CGMossa**
Rust added support for `[T; N]`, but we haven't taken advantage of it yet.

Previous attempts got scope creeped, thus this time we'll stick with a very small PR for now.

### CODE REVIEW

File: extendr-api/src/robj/into_robj.rs [hunk 1]
```diff
@@ -721,28 +704,49 @@ macro_rules! impl_from_as_iterator {
...
+}
+
+impl<T: ToVectorValue> From<Vec<T>> for Robj {
+    fn from(value: Vec<T>) -> Self {
+        let len = value.len();
+        fixed_size_collect(value.into_iter(), len)
+    }
+}
 
 impl_from_into_iter! {&'a [T]}
```
Comment (Ilia-Kosenkov) [inline extendr-api/src/robj/into_robj.rs:751]: If this is the only use of the macro now, should it be kept here, or maybe we can drop the macro in favor of just writing it out as is?

Comment (CGMossa) [reply extendr-api/src/robj/into_robj.rs:751]: I did make a PR where I tried to get rid of everything. Frankly, I'm just hoping for hyper scoped PRs, that get merged, and then I'll do whatever. But I'll take a look at this now.

Comment (CGMossa) [reply extendr-api/src/robj/into_robj.rs]: Oh this can be inlined as you say.. Oops.. I got rid of it now.

**Review by Ilia-Kosenkov — APPROVED** [review]
Looks clean & mich more readable.

**Pull request merged**